include/nbl/asset/utils/CSPIRVIntrospector.h

```diff
@@ -648,35 +648,30 @@ class NBL_API2 CSPIRVIntrospector : public core::Uncopyable
 			if (layout)
 			{
 				// TODO: merge CStageIntrospection into CPipelineIntrospection
+				pplnIntrospectData->merge(introspection.get());
 			}
 			else
 			{
 				pplnIntrospectData->merge(introspection.get());
 
-				auto pcRange = pplnIntrospectData->createPushConstantRangesFromIntrospection(introspection);
+				const auto pcRanges = pplnIntrospectData->createPushConstantRangesFromIntrospection(introspection);
+				const std::span<const asset::SPushConstantRange> pcRangesSpan = {
+					pcRanges ? pcRanges->begin() : nullptr,
+					pcRanges ? pcRanges->end() : nullptr
+				};
 
-				//layout = core::make_smart_refctd_ptr<ICPUPipelineLayout>(
-				//	pcRange, // TODO
-				//	pplnIntrospectData->createApproximateDescriptorSetLayoutFromIntrospection(0),
-				//	pplnIntrospectData->createApproximateDescriptorSetLayoutFromIntrospection(1),
-				//	pplnIntrospectData->createApproximateDescriptorSetLayoutFromIntrospection(2),
-				//	pplnIntrospectData->createApproximateDescriptorSetLayoutFromIntrospection(3)
-				//);
+				layout = core::make_smart_refctd_ptr<ICPUPipelineLayout>(
+					pcRangesSpan,
+					pplnIntrospectData->createApproximateDescriptorSetLayoutFromIntrospection(0),
+					pplnIntrospectData->createApproximateDescriptorSetLayoutFromIntrospection(1),
+					pplnIntrospectData->createApproximateDescriptorSetLayoutFromIntrospection(2),
+					pplnIntrospectData->createApproximateDescriptorSetLayoutFromIntrospection(3)
```

Comment:
we need to change the signature of `createApproximateDescriptorSetLayoutFromIntrospection` because not returning `nullptr` isnt really a criterion for success.

We should probably return a struct
```cpp
struct SSetLayoutCreateResult
{
  inline operator bool() const {return success;}

   core::smart_refctd_ptr<ICPUDescriptorSetLayout> layout;
   bool success;
};
```

then you can return a null layout if there are no bindings present with this set number and still count that as a success

Reply:
nvm what I wrote I'm dumb